### DISCUSSION TITLE
Conjuncts are not identified completely in spaCy

### DISCUSSION THREAD

**Discussion by Xinxin-Harbsafe2** [explosion/spaCy#8059] [Help: Other Questions] [answered]
Hi,

I have a problem while using the .conjuncts, i.e., not all conjuncts are identified. 

As described in the following link, anyone knows the reason? Thanks!

**Comment by polm** [comment, marked as answer]
Your question doesn't have any concrete examples. Can you give a specific example where you're not seeing the output you want, with sample code?

Based on your description I'm not sure if you're seeing a bug or not.

**Reply by Xinxin-Harbsafe2** [reply]
The code runs well, no error. However, as I have mentioned at the first time, it is just one pair of conjuncts cannot be identified while using `.conjuncts`, I do not know the reason.

**Reply by polm** [reply]
Looks perfectly fine to me?

```
import spacy

nlp = spacy.load("en_core_web_md")

text = 'software program that performs specific functions initiated by a user command or a process event and that can be executed without access to system control, monitoring, or administrative privileges'

doc = nlp(text)

for chunk in doc.noun_chunks:
    print("=====")
    print(chunk)
    print(chunk.conjuncts)
```

Output:

```
=====
software program
()
=====
specific functions
()
=====
a user command
(event,)
=====
a process event
(command,)
=====
access
()
=====
system control
(monitoring, privileges)
=====
monitoring
(control, privileges)
=====
administrative privileges
(control, monitoring)
```

Not sure why you're using `prev_end` instead of just using the chunk spans, is there a particular reason for that?

**Reply by Xinxin-Harbsafe2** [reply]
Thanks.

> why...using prev_end instead of just using the chunk spans...
Yes, I have a reason, because I want to split the sentence at the end of each chunk, but not only get the chunks.

I am wondering, why it does not identify the `a user command (event,)` and  `a process event(command,)` while using `prev_end`? 

Have you tried using the `prev_end`?
Thanks again!

**Reply by polm** [reply]
OK, here's code using prev_end. I also added the output of the root of each span so it's easier to see what's going on.

```
import spacy

nlp = spacy.load("en_core_web_md")

text = 'software program that performs specific functions initiated by a user command or a process event and that can be executed without access to system control, monitoring, or administrative privileges'

doc = nlp(text)

prev_end = 0
for chunk in doc.noun_chunks:
    print("=====")
    span = doc[prev_end:chunk.end]
    print(span, f"(root: {span.root})")
    print(span.conjuncts)
    prev_end = chunk.end
```

And here's the output:

```
=====
software program (root: program)
()
=====
that performs specific functions (root: performs)
(executed,)
=====
initiated by a user command (root: initiated)
()
=====
or a process event (root: or)
()
=====
and that can be executed without access (root: and)
()
=====
to system control (root: to)
()
=====
, monitoring (root: monitoring)
(control, privileges)
=====
, or administrative privileges (root: or)
()
```

If you look at the printed root in these spans, you can see that the head of the noun chunk is often not the root. Since `conjuncts` only returns conjuncts of the root of the span, it won't work the same as the test code using `noun_chunks` instead of `prev_end`. For example, in "initiated by a user command", the root is "initiated", so only conjuncts of that verb, not conjuncts of the noun "command", will be returned.

Does that make sense?

**Reply by Xinxin-Harbsafe2** [reply]
Thanks a lot! I should not use the ` conj_ = span.conjuncts`, I should use the ` conj_ = chunk.conjuncts` then. Thanks again!

**Reply by polm** [reply]
Note that the `chunk` is also a span - you just need to use a span where `span.root` is the token you want to get conjuncts for.
